docs/v3/codecs/blosc/v1.0.rst

```diff
@@ -70,33 +70,36 @@ clevel:
     level is 0.
 
 shuffle:
-    An integer value in the set {0, 1, 2, -1} indicating the way
-    bytes or bits are rearranged, which can lead to faster
-    and/or greater compression. A value of 1
-    indicates that byte-wise shuffling is performed prior to
-    compression. A value of 2 indicates the bit-wise shuffling is
-    performed prior to compression. If a value of -1 is given,
-    then default shuffling is used: bit-wise shuffling for buffers
-    with item size of 1 byte, byte-wise shuffling otherwise.
-    Shuffling is turned off completely when the value is 0.
+    Specifies the type of shuffling to perform, if any, prior to compression.
+    Must be one of:
+
+    - ``null``, to indicate no shuffling;
+    - ``"byte"``, to indicate byte-wise shuffling;
+    - ``"bit"``, to indicate bit-wise shuffling.
+
+typesize:
+    Positive integer specifying the stride in bytes over which shuffling is
+    performed.  Required unless ``shuffle`` is ``null``, in which case the value
+    is ignored.
```

Comment:
I'm not sure if this change is useful atm. I have a two of questions:
- Couldn't we make this optional, and have a auto-detect logic as before? It would only be necessary for dtypes with arbitrary length data.
- Why should it be called `typesize`? It could also be any valid divisor of the typesize, especially for arbitrary-length types, right?

I'd rather leave this out for now, and add it later once we have dtypes with arbitrary sizes. Then this can be an additional optional field. Although this would be nice:
> This allows Blosc to function as a pure "bytes -> bytes" codec rather than an "array -> bytes" codec.

Not quite sure if we should include this. cc @joshmoore @normanrz @rabernat @jakirkham What do you think about it?

Reply:
> If this is preferred I can try to write up the necessary changes to core specification.

I'd be in favor of this (Option 1), but would like to have some more opinions first. In my opinion this is blocking ZEP 1 / v3.

Reply:
No strong opinion here. Option 1 is fine with me. So is keeping `typesize` in. It only seems like a small redundancy. Might facilitate fallback dtype handling.